bash/maven_build.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+#/*******************************************************************************
+# * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+# *
+# * This program and the accompanying materials are made available under the
+# * terms of the Eclipse Public License 2.0 which is available at
+# * http://www.eclipse.org/legal/epl-2.0.
+# *
+# * SPDX-License-Identifier: EPL-2.0
+# *******************************************************************************/
+
+# --------------------------------------------------------------------------
+# This script handles the different calls to maven
+# --------------------------------------------------------------------------
+
+# fail the script if a call fails
+set -e
+
+# Store the name of the command calling from commandline to be properly
+# displayed in case of usage issues
+COMMAND=$0
+
+# this method gives some little usage info
+printUsage() {
+	echo "usage: ${COMMAND} -j [surefire|spotbugs|checkstyle|assemble] -p [development|integration|release]"
+	echo ""
+	echo "Options:"
+	echo " -j, --jobs <jobname>	    The name of the Travis-CI job to be build."
+	echo " -p, --profile <profile>  The name of the maven profile to be build."
+	echo ""
+	echo "Jobname:"
+	echo " surefire      To run all surefire tests including junit and swtbot."
+	echo " spotbugs      To run spotbugs static code analysis."
+	echo " checkstyle    To run checkstyle for testing style guidelines."
+	echo " assemble      To run full assemble including the java docs build."
+	echo ""
+	echo "Profile:"
+	echo " development      Maven profile for development and feature builds."
+	echo " integration      Maven profile for integration builds."
+	echo " release          Maven profile for release builds. Fails to overwrite deployments."
+	echo ""
+	echo "Copyright by DLR (German Aerospace Center)"
+}
+
+callMavenSurefire() {
+	echo "Maven - Surefire - ${MAVEN_PROFILE}"
+	mvn clean compile -P ${MAVEN_PROFILE},target -B -V | tee maven.log
+	echo "Check for Maven Problems on Overtarget:"
+	(grep -n "\[\(WARN\|ERROR\)\]" maven.log || exit 0  && exit 1;)
+	mvn install -P ${MAVEN_PROFILE},surefire,product -B -V | tee maven.log
+	echo "Check for Maven Problems on Product:"
+	(grep -n "\[\(WARN\|WARNING\|ERROR\)\]" maven.log || exit 0  && exit 1;)
+	echo "Check for failed test cases:"
+	(grep -n "<<< FAILURE!" maven.log || exit 0 && exit 1;)
+	echo "Ant jacoco Reports"
+	ant jacocoPrepareDependencies
+	ant jacocoReport 2>&1 | tee ant.log
+	(grep -n "Rule violated" ant.log || exit 0 && exit 1;)
```

Comment:
The build can also fail here if a e.g. a project does not have a source directory... Then surefire doesn't know how to handle the project. If that happens reports are not evaluated and the build succeeds without checking the coverage. "grep -n "Rule violated\|BUILD FAILED" ant.log" solves this. I have a fix for that on the requirements branch and can merge it later... Just to let you know ;)

Reply:
Good idea. I will include that suggestion.